rc4me/fetch_rcs.py

```diff
@@ -0,0 +1,101 @@
+"""Clone dotfile repo to rc4me home directory."""
+
+from pathlib import Path
+import logging
+import click
+import git
+
+logging.basicConfig(level=logging.DEBUG)
+logger = logging.getLogger(__name__)
+
+
+@click.command()
+@click.argument("rc4me_home", type=click.Path())
+@click.argument("repo", type=str)
+def _fetch_rcs_cli(rc4me_home: str, repo: str) -> None:
+    """CLI wrapper for fetch_rcs function.
+
+    Args:
+        rc4me_home: Location of rc4me home directory (e.g. ~/.rc4me).
+        repo: Local repo or GitHub repo stub (e.g. "jeffmm/vimrc").
+
+    Returns:
+        None
+
+    """
+    fetch_rcs(Path(rc4me_home), repo)
+
+
+def fetch_rcs(rc4me_home: Path, repo: str) -> None:
+    """Clone RC repository to local directory.
+
+    Clones rc4me repository to rc4me home directory at $HOME/.rc4me. If the
+    repo already exists locally, fetch updates and confirm update to origin
+    master. If the .rc4me home directory does not exist, scaffold it.
+
+    Args:
+        rc4me_home: Location of rc4me home directory (e.g. ~/.rc4me).
+        repo: Local repo or GitHub repo stub (e.g. "jeffmm/vimrc").
+
+    Returns:
+        None
+
+    """
+    repo_dir = rc4me_home / repo
+    # Check if the rc4me home directory exists, and if not, scaffold it
+    if not rc4me_home.exists():
+        init_rc4me(rc4me_home)
+    # First check whether the repo is already cloned in the home directory
+    if repo_dir.exists():
+        # If the repo already exists, fetch and update
+        fetch_info = git.Repo(repo_dir).remote("origin").fetch()
+        # Check that the repo is up to date
+        if fetch_info[0].commit.hexsha != git.Repo(repo_dir).head.commit.hexsha:
+            # Update on user confirmation
+            if _check_if_overwrite(repo):
+                git.Repo(repo_dir).remote("origin").pull("master")
+    # If it's not already in the home directory, check if the source directory refers to
+    # a local directory
+    elif Path(repo).exists():
```

Comment:
I see what's going on here. Clone from the local confused me, but seems reasonable to support